Now builds on stable.

thejpster opened the pull request:
Tested working with probe-rs CLI. Slightly larger binary, but I can live with that for not having to use nightly rust.

```console
$ ./build.sh
    Finished release [optimized + debuginfo] target(s) in 0.10s
    instructions: 8LUDr4mwK0x8RCB4ASgqTX1EKk5+RAPRKGiARzBogEcBIAiQIHAGlRglKogHlBQkIIgFlhtOMUb3MZBHBJAqiCCIGUmJHJBHA5AqiCCIMUaQRwKQKoggiBRJkEcBkCqIIIgTSZBHBkYqiCCIDkmQRwVGBJygRwOYgEcRSHhEApkBYBBIeEQEYA9IeEQBmQFgBpgGYAWYBWAImAeZCHAAIAmw8L1SRQAAQ1gAAFJQAABGQwAAeAEAAIABAACAAQAADAEAAAABAAACAQAA0LUCrwhMfEQgeAEoCtEHSHhEAGiARwZIeEQAaIBHACAgcNC9ASDQva4AAAC0AAAAsAAAAAVIeEQAeAAoAdEBIHBHAUhwR8BG0HAAAH4AAADQtQKvCUl5RAl4ASkM0Q8hCQdAGAZJeUQMaAEiEQMSBNgjoEcAINC9ASDQvV4AAABWAAAA0LUCrwtGCUl5RAl4ASkK0Q8hCQdAGAZJeUQMaBFGGkagRwAg0L0BINC9wEYoAAAAJAAAAADU1NQAAAAAAAAAAAAAAAAAAAAAAAAAAA==
    pc_init: 1
    pc_uninit: 205
    pc_program_page: 337
    pc_erase_sector: 285
    pc_erase_all: 257
```

It also uses the official ROM function for walking the ROM table entries rather than walking them manually.

thejpster committed:
Now builds on stable.

Tested working with probe-rs CLI. Slightly larger binary, but I can live with that for not having to use nightly rust.

jannic commented on src/main.rs:
```diff
@@ -3,28 +3,19 @@
+        let lookup_func: RomTableLookupFn = core::mem::transmute(lookup_func);
+        let table = core::ptr::read(0x0000_00014 as *const u16) as usize;
+        let result = lookup_func(table as *const u16, tag);
+        core::mem::transmute_copy(&result)
```
This will transmute 0 into T instead of panicking if an unknown tag is used. Just mentioning. I think it's fine, this function is very unsafe if used incorrectly anyway. If that was meant to be included in a larger code base, I'd ask for safety comments. But I think here it's ok. The whole program is using rust not as a safe language, but as a more convenient alternative to assembly.

thejpster replied:
That's a good point, and as the ROM contents are defined outside the program I should probably trap that.

jannic replied:
It would be cleaner, but the practical impact would be very small: If you use a flash algo on the wrong chip, chances for a satisfying result are very small either way.

thejpster replied:
The init function can return an error, so it makes sense to do that instead.

thejpster committed:
Fail if ROM funcs are missing.

Belt-and-braces in case we ever get a different RP2040 ROM with missing functions.

Also use constants for the ROM addresses.

thejpster committed:
Add docs so clippy is happy.
